src/securityPolicy.c

```diff
@@ -293,21 +302,39 @@ security_policy_t policyForSignTxOutputToken(
         token_amount_t* token MARK_UNUSED
 )
 {
-	ALLOW_IF(addressPolicy == POLICY_ALLOW_WITHOUT_PROMPT);
+	switch (addressPolicy) {
+	case POLICY_ALLOW_WITHOUT_PROMPT:
+		ALLOW();
 
-	SHOW();
+	case POLICY_SHOW_BEFORE_RESPONSE:
+		SHOW();
+
+	default:
+		ASSERT(false);
+	}
+
+	DENY(); // should not be reached
 }
 
 security_policy_t policyForSignTxOutputConfirm(
         security_policy_t addressPolicy,
         uint64_t numAssetGroups
 )
 {
-	ALLOW_IF(addressPolicy == POLICY_ALLOW_WITHOUT_PROMPT);
+	switch (addressPolicy) {
+	case POLICY_ALLOW_WITHOUT_PROMPT:
+		ALLOW();
+
+	case POLICY_SHOW_BEFORE_RESPONSE:
+		PROMPT_IF(numAssetGroups > 0);
+		ALLOW();
+
+	default:
+		ASSERT(false);
+	}
 
-	PROMPT_IF(numAssetGroups > 0);
+	DENY(); // should not be reached
```

Comment:
the comment is imho too self evident and doesn't add meaningful insight. We already do this in other part of the policies without explaining it